src/order/conditionally_complete_lattice.lean

```diff
@@ -557,6 +557,27 @@ begin
   apply lt_irrefl b (lt_of_le_of_lt ‹b ≤ Inf s› ‹Inf s < b›)
 end
 
+local attribute [instance] classical.prop_decidable
+
+/--Introduction rule to prove that b is the supremum of s: it suffices to check that
+1) b is an upper bound
+2) every other upper bound b' satisfies b ≤ b'.-/
+theorem cSup_intro' (_ : s ≠ ∅)
+  (_ : ∀ a ∈ s, a ≤ b) (H : ∀ub, (∀ a ∈ s, a ≤ ub) → (b ≤ ub)) : Sup s = b :=
+suffices ∀w, w < b → (∃a∈s, w < a), from cSup_intro ‹s ≠ ∅› ‹∀a∈s, a ≤ b› this,
```

Comment:
This should be a straight forward proof using `le_antisymm`, and staying in `≥ ` instead of going to `>`.

Reply:
Enlighten me ...

Reply:
Nice!